examples/plotting/plot_rectangle.py

```diff
@@ -0,0 +1,85 @@
+Drawing a Rectangle on a map
+==========================================================
+
+In this example we are going to look at how we can plot rectangles on maps
+while describing the dimensions using different methods.
+"""
+import matplotlib.pyplot as plt
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.data.sample
+import sunpy.map
+
+################################################################################
+# We use the `~sunpy.map.GenericMap.draw_rectangle` method to draw the rectangles.
+aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+fig = plt.figure(figsize=(5, 5))
+
+################################################################################
+# The first option is to provide an input `~astropy.coordinates.SkyCoord` as a
+# (2,)-shaped tuple in the form of ((x1, x2), (y1, y2)).
+fig.add_subplot(111, projection=aia_map)
+coords = SkyCoord((-400, 200) * u.arcsec, (-300, 100) * u.arcsec, frame=aia_map.coordinate_frame)
+aia_map.draw_rectangle(
+    coords,
+    color="blue",
+    linestyle="-",
+)
+aia_map.plot()
+plt.show()
+
+################################################################################
+# The second option is to pass `~astropy.coordinates.SkyCoord` bottom_left and
+# `~astropy.coordinates.SkyCoord` top_right.
+fig.add_subplot(111, projection=aia_map)
+bottom_left = SkyCoord(-400 * u.arcsec, -300 * u.arcsec, frame=aia_map.coordinate_frame)
+top_right = SkyCoord(300 * u.arcsec, 200 * u.arcsec, frame=aia_map.coordinate_frame)
+aia_map.draw_rectangle(
+    bottom_left,
+    top_right=top_right,
+    color="green",
+    linestyle="--",
+)
+aia_map.plot()
+plt.show()
+
+################################################################################
+# The third option is to provide `~astropy.coordinates.SkyCoord` bottom_left,
+# `~astropy.units.Quantity` width and `~astropy.units.Quantity` height.
+fig.add_subplot(111, projection=aia_map)
+bottom_left = SkyCoord(-500 * u.arcsec, -200 * u.arcsec, frame=aia_map.coordinate_frame)
+width = 700 * u.arcsec
+height = 500 * u.arcsec
+aia_map.draw_rectangle(
+    bottom_left,
+    width=width,
+    height=height,
+    color="yellow",
+    linestyle="-.",
+)
+aia_map.plot()
+plt.show()
+
+################################################################################
+# We also have a fourth option - to specify the dimensions in terms of pixel coordinates.
+#
+# To do this, we need to first create a `~sunpy.map.GenericMap.submap` by passing
+# `~astropy.units.Quantity` objects with the units as pixel. Then we can use the
+# `~sunpy.map.GenericMap.bottom_left_coord` and `~sunpy.map.GenericMap.top_right_coord` properties
+# as parameters for `~sunpy.map.GenericMap.draw_rectangle`.
+fig.add_subplot(111, projection=aia_map)
+bottom_left_pixel = [400, 400] * u.pixel
+top_right_pixel = [700, 600] * u.pixel
+aia_submap = aia_map.submap(bottom_left_pixel, top_right=top_right_pixel)
```

Comment:
Instead of using submap here I'd suggest its better to use the `sunpy.map.GenericMap.pixel_to_world` to get the coordinates of top right and bottom left 

e.g.
```python
bottom_left_coord = aia_map.pixel_to_world(400*u.pixel, 400*u.pixel)
top_right_coord = aia_map.pixel_to_world(700*u.pixel, 600*u.pixel)
aia_map.draw_rectangle(bottom_left_coord, top_right=top_right_coord)
```

Reply:
Done, thanks!